pkg/cluster/nodes/nodes.go

```diff
@@ -99,3 +100,35 @@ func list(visit func(string, *Node), filters ...string) error {
 	}
 	return nil
 }
+
+// WaitForReady uses kubectl inside the "node" container to check if the
+// master nodes are "Ready".
+func WaitForReady(node *Node, until time.Time) bool {
+	return tryUntil(until, func() bool {
+		cmd := node.Command(
+			"kubectl",
+			"--kubeconfig=/etc/kubernetes/admin.conf",
+			"get",
+			"nodes",
+			"--selector=node-role.kubernetes.io/master",
+			// When the node reaches status ready, the status field will be set
+			// to true.
+			"-o=jsonpath='{.items..status.conditions[-1:].status}'",
+		)
+		lines, err := exec.CombinedOutputLines(cmd)
+		if err != nil {
+			return false
+		}
+		if len(lines) != 1 {
+			return false
+		}
+
+		// Check node status. The jsonpath output will come with quotes around
+		// the status.
+		if lines[0] == "'True'" {
+			return true
+		}
+
```

Comment:
it might be worth adding a small time.Sleep() duration in the function. 1 second seems OK to me.